planner/core/rule_eliminate_projection.go

```diff
@@ -148,6 +159,22 @@ func (pe *projectionEliminater) eliminate(p LogicalPlan, replace map[string]*exp
 	return p.Children()[0]
 }
 
+func replaceColumnOfExpr(expr expression.Expression, proj *LogicalProjection) expression.Expression {
+	switch v := expr.(type) {
+	case *expression.Column:
+		for i := range proj.Schema().Columns {
```

Comment:
It seems we can use `proj.Schema().ColumnIndex(v)`?

Reply:
Fixed

Reply:
Could you please add a SQl to reproduce it?
`HashCode()` just encode the `UniqueID`. So compare hashcode should be same with `Equal()`.